app/models/info_request/sluggable.rb

```diff
@@ -33,25 +33,26 @@ def url_title(opts = {})
 
   def update_url_title
     return unless title
+
     url_title = MySociety::Format.simplify_url_part(title, 'request', 32)
-    conditions = id ? ["id <> ?", id] : []
-
-    existing_url_titles = InfoRequest.
-      where(conditions).
-      where('url_title ~ ?', "^#{url_title}(_\\d+)?$").
-      pluck(:url_title)
-
-    if existing_url_titles.empty?
-      unique_url_title = url_title
-    else
-      # For request with same title as others, add on arbitrary numeric
-      # identifier
-      suffix_num = existing_url_titles.map { |t| (t[/_(\d+)$/, 1] || 1).to_i }.
-                                       sort.
-                                       last + 1
-      unique_url_title = "#{url_title}_#{suffix_num}"
-    end
+    suffix = suffix_number(url_title)
+    unique_url_title = suffix ? "#{url_title}_#{suffix}" : url_title
 
     write_attribute(:url_title, unique_url_title)
   end
+
+  def suffix_number(url_title)
+    # For request with same title as others, add on arbitrary numeric identifier
+    suffixes = existing_url_titles(url_title)
+    suffixes.last + 1 if suffixes.present?
+  end
+
+  def existing_url_titles(url_title)
+    scope = InfoRequest
+    scope = scope.where.not(id: id) if persisted?
```

Comment:
minor nitpick, I'd be tempted to restate this as:

```rb
    scope =
      if persisted?
        InfoRequest.where.not(id: id)
      else
        InfoRequest
      end

```

arguably (as in "feel free to argue"!) clearer and might be a tiny fraction faster without assigning `scope` twice (although a quick Benchmark test suggests we're literally talking a couple of thousandths of a second so ... meh)